### ISSUE TITLE
Improvements for the color contrast check against default theme colors.

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#6484]
Premise: I haven't looked at the code.

When setting the text color to a value that produces an insufficient contrast ratio with the default background color, there's no warning:

Custom text color vs. default background (white in this case):

![screen shot 2018-04-28 at 16 09 04](https://user-images.githubusercontent.com/1682452/39397421-3275db70-4aff-11e8-9612-f379b4583bb6.jpg)

Vice-versa, a custom background that produces an insufficient contrast ratio with the default text color, does show the warning:

![screen shot 2018-04-28 at 16 09 45](https://user-images.githubusercontent.com/1682452/39397436-59564310-4aff-11e8-92c9-3b3a1b6d1f28.jpg)

It would be great to have the check work also with the default background color. 

Also wondering what happens when in the future themes will be able to provide their own background / text colors, see for example #6404:
> One long term vision for the editor, is to allow you to load the theme stylesheet into the editor canvas itself ...

**Comment by jorgefilipecosta** [comment]
Hi @afercia, what happens is that by default the background of the paragraph is rgba( 0, 0, 0, 0) (totally transparent), we can not know what is, in fact, the background color. The theme may even set an image as background and we can not compute the right color. If the theme sets a background color e.g: white and correctly loads the styles, we will be able to shown warnings. But if no changes happen we can not make sure if the contrast is right or not.

For the text color, we show warning because at the start it is not transparent, by default we set a color.

**Comment by afercia** [comment]
Hi @jorgefilipecosta thanks, I understand a bit better now. Then if there are no reliable ways to detect the background color when blocks are transparent and the theme doesn't set a background color, please do feel free to close this issue,

**Comment by jorgefilipecosta** [comment]
Unfortunately, it looks like in fact, right now we don't have a reliable way to detect what the background color (or colors) will end up being. So I'm closing this issue, for now, thank you for referring this issue.

**Comment by afercia** [comment]
For some background, see https://github.com/WordPress/gutenberg/pull/5658#discussion_r176880626
